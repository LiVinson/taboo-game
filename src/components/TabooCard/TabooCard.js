import React from "react"
import "./TabooCard.scss"
import ButtonGroup from "../ButtonGroup"

export default function TabooCard({ type, tabooWord, list, children, buttonObj=null }) {


  let secondaryClass = null
  if (type) {
    secondaryClass = `taboo-card--${type}`
  }
  return (
      <div className={`taboo-card ${secondaryClass}`}>
        <div className="taboo-card__word-container">
          <p className="taboo-card__word">{tabooWord}</p>
        </div>
        <div className="taboo-card__list-container">
        {children ? children :
          <ul className="taboo-card__list">
          {list.map((item,index) => <li key={index}className="taboo-card__list-item">{item}</li>)}
        </ul>
        }
        
          <ButtonGroup buttonInfo={buttonObj} />
         
        </div>
     </div>

  )
}


/*

  <div className={`taboo-card ${secondaryClass}`}>
        <div className="taboo-card__word-container">
          <p className="taboo-card__word">Taboo!</p>
        </div>
        <div className="taboo-card__list-container">
          <ul className="taboo-card__list">
            <li className="taboo-card__list-item">Taboo!</li>
            <li className="taboo-card__list-item">Taboo!</li>
            <li className="taboo-card__list-item">Taboo!</li>
            <li className="taboo-card__list-item">Taboo!</li>
          </ul>
        </div>
      </div>

      <div className={`taboo-card ${secondaryClass}`}>
        <div className="taboo-card__word-container">
          <p className="taboo-card__word">Taboo!</p>
        </div>
        <div className="taboo-card__list-container">
          <ul className="taboo-card__list">
            <li className="taboo-card__list-item">Taboo!</li>
            <li className="taboo-card__list-item">Taboo!</li>
            <li className="taboo-card__list-item">Taboo!</li>
            <li className="taboo-card__list-item">Taboo!</li>
          </ul>
        </div>
      </div>*/