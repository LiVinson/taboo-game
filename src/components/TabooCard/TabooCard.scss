.taboo-card {
    width:30rem; 
    margin: 0 auto;
    padding: 2rem;
    border-radius: 2px;
    position: relative;
    z-index: 15;
    background-color: var(--color-primary);

    box-shadow: 0px .8rem 1rem rgba(0,0,0,.9);
    //Used for inheritance on child components
    color: var(--color-primary-light);
    


    //Main taboo card on home screen, make text and padding larger
    &--home &__word-container &__word {
        font-size: 4rem;
    }

    &--home &__list-container {
        padding-top:2rem;
        padding-bottom:2rem;
    }

    &--bottom {    
        color:  var( --color-tertiary);
        background-color: currentColor;
        position: absolute;
        bottom:9rem;
        right: 8rem;
        z-index: 5;
        transform: rotate(-42deg);
    }

    &--middle {
        color:  var( --color-secondary);
        background-color: currentColor;
        position: absolute;
        bottom:8rem;
        left:7rem;
        z-index: 10;
        transform: rotate(42deg)
    }

    &__word-container
    {
        background-color: var(--color-off-white);
        padding: 1.2rem;
        text-align: center;   
        margin-bottom: 1.5rem;
        border-radius: 2px;
        color:inherit;

    }

    &__word {
        font-family: 'Oswald', sans-serif;
        font-size: 3.5rem;
        color: inherit;
        font-weight: 600;
        text-transform: uppercase;
    
    }

    &__list-container {
        background-color: var(--color-off-white);
        padding: 1.5rem;
        text-align: center;  
        border-radius: 2px;
    }

    &__list {
        list-style-type: none;

    }

    &__list-item {
        font-family: 'Oswald', sans-serif;
        font-size: 2.5rem;
        position: relative;
        padding-bottom:1.5rem;
    }


    &__list-item:after {
        content:"";
        display:block;
        width: 10rem;
        margin: 0 auto;
        padding-top:1rem;
        border-bottom: #C4C4C4 2px solid; 
    }
}