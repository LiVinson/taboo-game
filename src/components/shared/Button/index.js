import React from "react"
import { StyledButton, StyledPrimaryButton, StyledLargeButton, StyledTextButton} from "./style"

//Takes in object of properties passed down. Typically includes type, onClick, text, and any other required attributes
export const Button = (props) => {
  return <StyledButton {...props}>{props.text}</StyledButton>
}


export const PrimaryButton = (props) => {
  return <StyledPrimaryButton {...props}>{props.text}</StyledPrimaryButton>
}

export const LargeButton = (props) => {
  return <StyledLargeButton {...props}>{props.text}</StyledLargeButton>
}

export const TextButton = (props) => {
  return <StyledTextButton {...props}>{props.text}</StyledTextButton>
}