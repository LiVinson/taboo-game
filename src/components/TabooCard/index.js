import TabooCard, { TabooCardTop } from "./TabooCard"

export { TabooCardTop };
export default TabooCard;
