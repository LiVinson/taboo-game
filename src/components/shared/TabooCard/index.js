		//If item is an object, filter based on property  value. Otherwise filter based on value of item (i.e. string)
import { TabooCard, ButtonTabooCard, LayeredTabooCard, TabooCardTop, FilteredTabooList} from "./TabooCard"

		//If item is an object, filter based on property  value. Otherwise filter based on value of item (i.e. string)
export { TabooCard, ButtonTabooCard,  LayeredTabooCard, TabooCardTop, FilteredTabooList};

