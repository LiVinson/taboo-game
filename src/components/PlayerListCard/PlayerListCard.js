import React from 'react'
import PropTypes from 'prop-types'
import { ButtonTabooCard, TabooCard } from 'components/shared/TabooCard'

const PlayerListCard = ({ buttonInfo, tabooWord, children }) => {
	return buttonInfo ? (
		<ButtonTabooCard tabooWord={tabooWord} buttons={buttonInfo}>
			{children}
		</ButtonTabooCard>
	) : (
		<TabooCard tabooWord={tabooWord}>{children}</TabooCard>
	)
}

PlayerListCard.defaultProps = {
	tabooWord: "Players"
}

PlayerListCard.propTypes = {
	buttonInfo: PropTypes.array,
	tabooWord: PropTypes.string.isRequired,
	children: PropTypes.node.isRequired

}

export default PlayerListCard
