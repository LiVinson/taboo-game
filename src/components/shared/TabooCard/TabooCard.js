import React from 'react'
import PropTypes from 'prop-types'
import {
	StyledTabooCard,
	StyledLayeredTabooCard,
	TabooWordContainer,
	TabooBody,
	StyledListItem,
	StyledListItemSecondary,
	StyledListTitle,
} from './style'
import List from 'components/shared/List'
import ButtonContainer from 'components/ButtonContainer'
import Star from 'components/shared/Star'
import {ErrorMessage} from 'components/shared/FeedbackMessage'

//Primary TabooCard component used for main menu and game
export const TabooCard = ({ tabooWord, list, children }) => {
	return (
		<StyledTabooCard>
			<TabooWordContainer>{tabooWord}</TabooWordContainer>
			<TabooBody>{children ? children : <TabooList list={list} />}</TabooBody>
		</StyledTabooCard>
	)
}

TabooCard.propTypes = {
	tabooWord: PropTypes.string.isRequired,
	list: PropTypes.array,
	children: PropTypes.node,
}

//Same as TabooCard, but includes 2 -3 buttons
export const ButtonTabooCard = ({ tabooWord, list, buttons, children, error }) => {
	return (
		<StyledTabooCard>
			<TabooWordContainer>{tabooWord}</TabooWordContainer>
			<TabooBody>
				{children ? children : <TabooList list={list} />}
				{error && <ErrorMessage error={error} />}
			</TabooBody>
			<ButtonContainer buttons={buttons} />
		</StyledTabooCard>
	)
}

ButtonTabooCard.propTypes = {
	tabooWord: PropTypes.string.isRequired,
	buttons: PropTypes.array.isRequired,
	list: PropTypes.array,
	children: PropTypes.node,
}

//Used for background element
export const LayeredTabooCard = ({ tabooWord, list, layer }) => {
	return (
		<StyledLayeredTabooCard layer={layer}>
			<TabooWordContainer>{tabooWord}</TabooWordContainer>
			<TabooBody>
				<TabooList list={list} />
			</TabooBody>
		</StyledLayeredTabooCard>
	)
}

LayeredTabooCard.propTypes = {
	tabooWord: PropTypes.string.isRequired,
	list: PropTypes.array.isRequired,
	layer: PropTypes.string.isRequired,
}

//Used in TabooCard, ButtonTabooCard and FilteredListTabooCard
const TabooList = ({ list, property, specialValue, specialKey, secondary }) => {
	return (
		<List>
			{list.map((listItem, index) => {
				let displayText
				let includeStar = false
				if (typeof listItem === 'object' && listItem !== null && listItem[property]) {
					displayText = listItem[property]

					if (listItem[specialKey] && listItem[specialKey] === specialValue) {
								includeStar = true
					}
				} else {
					displayText = listItem
				}
				return (
					<TabooListItem listItem={displayText} key={index} includeStar={includeStar} secondary={secondary} />
				)
			})}
		</List>
	)
}

TabooList.propTypes = {
	list: PropTypes.array.isRequired,
	property: PropTypes.string,
	specialValue: PropTypes.string,
	specialKey: PropTypes.string,
	secondary: PropTypes.bool,
}

//exported for use in tests
export const TabooListItem = ({ listItem, includeStar, secondary }) => {
	//secondary styling and option to include Star
	return secondary ? (
		<StyledListItemSecondary>
			{listItem}
			{includeStar && <Star />}
		</StyledListItemSecondary>
	) : (
		<StyledListItem>{listItem}</StyledListItem>
	)
}

TabooListItem.propTypes = {
	listItem: PropTypes.any,
	includeStar: PropTypes.bool,
	secondary: PropTypes.bool,
}

export const TabooCardTop = ({ children, margin = false }) => {
	return (
		<StyledTabooCard margin={margin}>
			<TabooWordContainer>{children}</TabooWordContainer>
		</StyledTabooCard>
	)
}

TabooCardTop.propTypes = {
	// children: PropTypes.string.isRequired,
	margin: PropTypes.bool,
}

//ex: playerList array of player objects need to be filtered by team: playerList, team, team1, name, Players, [{}]
export const FilteredTabooList = ({
	unfilteredList,
	listTitle,
	filterKey,
	filterValue,
	displayProperty,
	specialKey,
	specialValue,
	noneMessage,
}) => {

	//filters array based on match of item or item[filterKey] equal to filterValue
	const filteredList = unfilteredList.filter((item) => {
		//If item is an object, filter based on property  value. Otherwise filter based on value of item (i.e. string)
		if (typeof item === 'object' && item !== null) return item[filterKey] === filterValue
		return item === filterValue
	})

	//Displays List of filtered values based on the displayProperty
	return (
		<>
			<StyledListTitle>{listTitle}</StyledListTitle>
			{filteredList.length > 0 ? (
				<TabooList
					list={filteredList}
					property={displayProperty}
					specialKey={specialKey}
					specialValue={specialValue}
					secondary={true}
				/>
			) : (
				<TabooList list={[noneMessage]} secondary={true} />
			)}
		</>
	)
}
