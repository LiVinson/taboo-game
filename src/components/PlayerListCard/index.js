import PlayerListCard from "./PlayerListCard"

export default PlayerListCard