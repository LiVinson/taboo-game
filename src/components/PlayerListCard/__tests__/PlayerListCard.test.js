import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import PlayerListCard from '../PlayerListCard'
import { ButtonTabooCard, TabooCard } from 'components/shared/TabooCard'

describe('PlayerLIstCard render and functionality is correct', () => {
	const defaultProps = {
		buttonInfo: [
			{
				text: 'Team 1',
				value: 'team 1',
				onClick: jest.fn(),
			},
			{
				text: 'Team 2',
				value: 'team 2',
				onClick: jest.fn(),
			},
		],
		tabooWord: 'Players',
	}

	const value = (
		<ul>
			<li>Child 1</li>
			<li>Child 2</li>
		</ul>
	)

	test('renders correctly', () => {
		const props = {
			...defaultProps,
		}

		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<PlayerListCard {...props}>{value}</PlayerListCard>
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('renders ButtonTabooCard when buttons prop provided', () => {
		const props = {
			...defaultProps,
		}
		const component = shallow(<PlayerListCard {...props}>{value}</PlayerListCard>)
		expect(component.find(ButtonTabooCard)).toHaveLength(1)
		expect(component.find(TabooCard)).toHaveLength(0)
	})

	test('renders TabooCard when buttons prop not provided', () => {
		const props = {
			tabooWord: 'Players'
		}

		const component = shallow(<PlayerListCard {...props}>{value}</PlayerListCard>)
		expect(component.find(ButtonTabooCard)).toHaveLength(0)
		expect(component.find(TabooCard)).toHaveLength(1)
	})
})
