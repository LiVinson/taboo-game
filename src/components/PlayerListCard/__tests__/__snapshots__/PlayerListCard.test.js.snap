// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerLIstCard render and functionality is correct renders correctly 1`] = `
<div
  className="sc-AxjAm hDlFTS"
>
  <div
    className="sc-AxiKw dGnLMo"
  >
    Players
  </div>
  <div
    className="sc-AxhCb JZaCe"
  >
    <ul>
      <li>
        Child 1
      </li>
      <li>
        Child 2
      </li>
    </ul>
  </div>
  <div
    className="sc-fzplWN buDbqD"
  >
    <button
      className="sc-AxmLO gmmVt"
      onClick={[MockFunction]}
      value="team 1"
    >
      Team 1
    </button>
    <button
      className="sc-AxmLO sc-fzozJi jlLOkV"
      onClick={[MockFunction]}
      value="team 2"
    >
      Team 2
    </button>
  </div>
</div>
`;
